pkg/algorithm/nodeselect/v1alpha1/select_node.go

```diff
@@ -186,6 +159,46 @@ func (ac *Config) selectNode(nodeBlockDeviceMap map[string]*blockDeviceList) *no
 	// minRequiredDiskCount will hold the required number of disk that should be selected from a qualified
 	// node for specific pool type
 	minRequiredDiskCount := spcv1alpha1.DefaultDiskCount[ac.poolType()]
+
+	if ProvisioningType(ac.Spc) == ProvisioningTypeAuto {
+		// List all blockdeviceclaims related to spc
+		// NOTE: This list contains blockdevices on which csps' are already
+		// created filter and use it
+		bdcList, err := bdc.NewKubeClient().
+			WithNamespace(ac.Namespace).
+			List(metav1.ListOptions{LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + ac.Spc.Name})
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to select node and blockdevices")
+		}
+		// customBDCList is a wrapper over blockdeviceclaim list
+		customBDCList := bdc.BlockDeviceClaimList{
+			ObjectList: bdcList,
+		}
+		// Form node and blockdevice topology for blockdevices which has bdc
+		claimedDevicesOnNode := customBDCList.GetBlockDeviceNamesByNode()
+		for claimedNodeName, bdNameList := range claimedDevicesOnNode {
+			qualifiedBDList := []string{}
+			filteredBlockDevices := nodeBlockDeviceMap[claimedNodeName]
+
+			//Check is this node is usable or not
+			if filteredBlockDevices != nil &&
+				util.ContainsString(filteredBlockDevices.Items, bdNameList[0]) {
+				qualifiedBDList = append(qualifiedBDList, bdNameList...)
+				if len(qualifiedBDList) < minRequiredDiskCount {
+					count := minRequiredDiskCount - len(qualifiedBDList)
+					availableBDOnNode := util.ListDiff(filteredBlockDevices.Items, qualifiedBDList)
+					if len(availableBDOnNode) < count {
+						continue
+					}
+					qualifiedBDList = append(qualifiedBDList, availableBDOnNode[:count]...)
+				}
+				selectedBlockDevice.BlockDevices.Items = append(selectedBlockDevice.BlockDevices.Items, qualifiedBDList...)
```

Comment:
I think - this causes problem. Here, we are selecting disks which can be out of filterBlockDevices. Can you please look into this?
this is similar to the issue that seen previously.

Reply:
if I'm not wrong, this need to be intersection of filteredBlockDevices and bdNameList

Reply:
No It won't because we are checking whether it can be usable or not by looking into filteredBlockdeviceList.
Link of check
https://github.com/openebs/maya/blob/d2ea21dc4433b51ee6d559bb96c251101316006b/pkg/algorithm/nodeselect/v1alpha1/select_node.go#L185

Reply:
we are checking for only first disk

Reply:
I think this is good enough because we are forming node and blockdevice topology checking on one BD is good to enough to exclude it from creating pool again.

Reply:
>  If I'm not wrong, this need to be intersection of filteredBlockDevices and bdNameList

In partial claims creations case intersection won't helpful we will get bd count less than required bd count